### DISCUSSION TITLE
Setting gradient attributes to `Trainer` when using manual optimization

### DISCUSSION THREAD

**Discussion by clement-pages** [Lightning-AI/pytorch-lightning#20463] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hey lightning team,

I am currently working on adding manual optimization to `pyannote.audio` in this [pull request](https://github.com/pyannote/pyannote-audio/pull/1796). The goal is to let the user choose between manual and automatic optimization by setting `model.automatic_optimization`. In our code, we want to retrieve gradient attributes (for instance `gradient_clip_val` or `gradient_clip_algorithm`) from the trainer by doing `model.trainer.gradient_clip_val` for example.

The problem is that we cannot instanciate a `Trainer` object with `gradient_clip_val` if we set `automatic_optimization` to `False`. Lightning throws the following exception:
```
lightning_fabric.utilities.exceptions.MisconfigurationException: Automatic gradient clipping is not supported for manual optimization. Remove `Trainer(gradient_clip_val=5.0)` or switch to automatic optimization.
```

So my question is : Is there any way to instanciate a trainer with gradient attributes and manual optimization without rising this error ? Maybe it could be a warning message instead of an error ? I have found a work around, but not having this error would clearly simplify the code.

Have a nice day!
